x-pack/plugin/security/qa/tls-basic/src/javaRestTest/java/org/elasticsearch/xpack/security/TlsWithBasicLicenseIT.java

```diff
@@ -34,6 +37,18 @@
 public class TlsWithBasicLicenseIT extends ESRestTestCase {
     private static Path httpTrustStore;
 
+    @Override
+    protected RestClient buildClient(Settings settings, HttpHost[] hosts) throws IOException {
+        RestClientBuilder builder = RestClient.builder(hosts);
+        configureClient(builder, settings);
+        if (settings.hasValue("xpack.security.enabled")) {
+            builder.setStrictDeprecationMode(true);
+        } else {
+            builder.setStrictDeprecationMode(false);
+        }
```

Comment:
If security is not set explicitly (some tests do this) I want client strict deprecation mode to be false to ignore the header, however if security is being set, the client should be in the strict deprecation mode. It may be not needed for this specific test at this time, but it is more generic. For example, `EnableSecurityOnBasicLicenseIT` runs with and w/o security being set, so the client will check for warning headers in the first case and will not in the second.

Reply:
Thank you for the explanation. A couple of points still:

* Is `xpack.security.enabled` a valid client setting? Can you please point me to a test that sets that?
* Ignoring all warning headers because we're expecting a particular one is not great. I'd rather expose what's required to override in such cases, if possible.
* Even if ignoring all headers is the sole option, we should contain this to the specific test/projects. This whole PR is being specific about security enable status so I imagine there are very few cases where we left it implicit. I don't think this is the case for `TlsWithBasicLicenseIT` class.

Reply:
- Mea culpa! Meant: `System.getProperty("tests.has_security") != null` Thanks for noticing!
- We (`es`) have some subset of tests ignoring warning headers. With this being said, I agree that we shouldn't ignore all the warning. I added it to two specific test cases where security is not set. Once security is on by defauld, it shouldn't be the problem anymore.
- Agree! Updated the Tls test to use `security`